toadr3/report.py

```diff
@@ -125,3 +127,113 @@ def payload_descriptors(self) -> list[ReportPayloadDescriptor] | None:
     def resources(self) -> list[ReportData]:
         """The resources of the report."""
         return self._resources
+
+
+def create_report(
+    event: Event,
+    client_name: str,
+    report_type: str,
+    report_values: list[Any],
+    report_name: str | None = None,
+) -> Report:
+    """Create a new report object.
+
+    This will create a single report object for the given event for the specific report types.
+    There are several assumptions made in this function:
+    1. You want/can only create a report for one report descriptor (at a time).
+    2. There is only one target the report applies to.
+    3. There is only one interval, and it is specified in an 'intervalPeriod' at the root level.
+
+    Parameters
+    ----------
+    event : Event
+        The event object to create the report for.
+    client_name : str
+        The client name for the report.
+    report_type : str
+        The report type for the report (payload type of one of report descriptors).
+    report_values : Any
+        The report values for the report.
+    report_name : str, optional
+        The name for the report (for debugging).
+
+    Raises
+    ------
+    ValueError
+        For missing and invalid arguments.
+
+    Returns
+    -------
+    Report
+        The created report object.
+    """
+    if client_name is None:
+        raise ValueError("client_name is required.")
+
+    if event.program_id is None:
+        raise ValueError("event must have a program ID.")
+
+    if event.id is None:
+        raise ValueError("event must have an ID.")
+
+    if event.interval_period is None:
+        raise ValueError("event must have an interval period.")
+
+    data = {
+        "objectType": "REPORT",
+        "programID": event.program_id,
+        "eventID": event.id,
+        "clientName": client_name,
+    }
+
+    if report_name is not None:
+        data["reportName"] = report_name
+
+    if event.report_descriptors is None or len(event.report_descriptors) == 0:
+        raise ValueError("event does not have any report_descriptors.")
+
+    report_descriptor = None
+    for rd in event.report_descriptors:
```

Comment:
Can multiple report descriptors match?

Reply:
Maybe. This is part of the assumptions caveat for this PR. :)